diff --git a/partitionmanager.cpp b/partitionmanager.cpp
index ca7e232..3c4a77f 100755
--- a/partitionmanager.cpp
+++ b/partitionmanager.cpp
@@ -494,8 +494,10 @@ void TWPartitionManager::Setup_Fstab_Partitions(bool Display_Error) {
 			}
 		}
 	#ifndef USE_VENDOR_LIBS
+#ifndef TW_KEEP_VENDOR_MOUNTED_FOR_CRYPTO
 		if (ven)
 			ven->UnMount(Display_Error);
+#endif
 		if (sys)
 			sys->UnMount(Display_Error);
 	#endif
@@ -539,6 +541,14 @@ void TWPartitionManager::Setup_Fstab_Partitions(bool Display_Error) {
 			Setup_Settings_Storage_Partition(settings_partition);
 		}
 
+#ifdef TW_KEEP_VENDOR_MOUNTED_FOR_CRYPTO
+		if (ven && !ven->Is_Mounted()) ven->Mount(Display_Error);
+		if (ven && ven->Is_Mounted()) {
+			property_set("twrp.vendor.mounted", "1");
+		} else {
+			LOGERR("Unable to mount vendor for crypto HAL services\n");
+		}
+#endif
 	#ifdef TW_INCLUDE_CRYPTO
 		DataManager::SetValue(TW_IS_ENCRYPTED, 1);
 		Decrypt_Data();
@@ -2388,7 +2398,9 @@ void TWPartitionManager::UnMount_Main_Partitions(void) {
 
 	TWPartition *Partition = Find_Partition_By_Path ("/vendor");
 
+#ifndef TW_KEEP_VENDOR_MOUNTED_FOR_CRYPTO
 	if (Partition != NULL) UnMount_By_Path("/vendor", false);
+#endif
 	UnMount_By_Path (Get_Android_Root_Path(), true);
 	Partition = Find_Partition_By_Path ("/product");
 	if (Partition != NULL) UnMount_By_Path("/product", false);
