diff --git a/keystore2/src/keystore2_main.rs b/keystore2/src/keystore2_main.rs
--- a/keystore2/src/keystore2_main.rs
+++ b/keystore2/src/keystore2_main.rs
@@ -20,6 +20,6 @@ use keystore2::maintenance::Maintenance;
 use keystore2::metrics::Metrics;
 use keystore2::metrics_store;
 use keystore2::service::KeystoreService;
-use keystore2::{apc::ApcManager, shared_secret_negotiation};
+use keystore2::apc::ApcManager;
 use keystore2::{authorization::AuthorizationManager, id_rotation::IdRotationState};
 use legacykeystore::LegacyKeystore;
@@ -92,6 +92,5 @@ fn main() {
     ENFORCEMENTS.install_confirmation_token_receiver(confirmation_token_receiver);
 
     entropy::register_feeder();
-    shared_secret_negotiation::perform_shared_secret_negotiation();
 
     info!("Starting thread pool now.");
