diff --git a/keystore2/src/globals.rs b/keystore2/src/globals.rs
--- a/keystore2/src/globals.rs
+++ b/keystore2/src/globals.rs
@@ -229,4 +229,9 @@ fn connect_keymint(
         (km, Some(km_version * 100))
     } else {
+        if *security_level == SecurityLevel::STRONGBOX {
+            return Err(Error::Km(ErrorCode::HARDWARE_TYPE_UNAVAILABLE)).context(ks_err!(
+                "No declared StrongBox KeyMint service in recovery."
+            ));
+        }
         // This is a no-op if it was called before.
         keystore2_km_compat::add_keymint_device_service();
