diff --git a/Android.bp b/Android.bp
--- a/Android.bp
+++ b/Android.bp
@@ -39,6 +39,7 @@ cc_defaults {
         "android.hardware.boot@1.0",
         "android.hardware.boot-V1-ndk",
         "android.hardware.keymaster@4.1",
+        "android.hardware.weaver-V2-ndk",
         "android.hardware.security.keymint-V3-ndk",
         "android.security.authorization-ndk",
         "libboot_control_client",
