diff --git a/libtar/Android.mk b/libtar/Android.mk
--- a/libtar/Android.mk
+++ b/libtar/Android.mk
@@ -28,6 +28,7 @@ ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
         android.hardware.keymaster@4.0 \
         android.hardware.keymaster@4.1 \
         android.hardware.weaver@1.0 \
+        android.hardware.weaver-V2-ndk \
         android.security.apc-ndk \
         android.system.keystore2-V1-ndk \
         android.security.authorization-ndk \
