diff --git a/keystore2/src/globals.rs b/keystore2/src/globals.rs
--- a/keystore2/src/globals.rs
+++ b/keystore2/src/globals.rs
@@ -252,11 +252,11 @@ fn connect_keymint(
 
     // If the KeyMint device is back-level, use a wrapper that intercepts and
     // emulates things that are not supported by the hardware.
     let keymint = match hal_version {
-        Some(300) => {
-            // Current KeyMint version: use as-is as v3 Keymint is current version
+        Some(v) if v >= 300 => {
+            // Current-or-newer KeyMint version: use as-is.
             log::info!(
-                "KeyMint device is current version ({:?}) for security level: {:?}",
+                "KeyMint device is current-or-newer version ({:?}) for security level: {:?}",
                 hal_version,
                 security_level
             );
