diff --git a/Android.bp b/Android.bp
index e010bf1..f339783 100755
--- a/Android.bp
+++ b/Android.bp
@@ -66,6 +66,7 @@ cc_library_static {
         "//external/puffin:__subpackages__",
         "//packages/modules/adb:__subpackages__",
         "//system/core/fs_mgr/libsnapshot:__subpackages__",
+        "//system/extras/partition_tools:__subpackages__",
         "//system/update_engine:__subpackages__",
     ],
 
